### PR TITLE
ARROW-5146: [Dev] Fix project name inference in merge script

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
@kou @xhochy Any objection?

File: dev/merge_arrow_pr.py [hunk 1]
```diff
@@ -378,9 +378,8 @@ def merge(self, target_ref='master'):
 
 def cli():
     # Location of your Arrow git clone
-    SEP = os.path.sep
-    ARROW_HOME = os.path.abspath(__file__).rsplit(SEP, 2)[0]
-    PROJECT_NAME = ARROW_HOME.rsplit(SEP, 1)[1]
+    ARROW_HOME = os.path.abspath(os.path.dirname(__file__))
+    PROJECT_NAME = os.environ.get('ARROW_PROJECT_NAME', '') or 'arrow'
```
Comment (kou) [inline dev/merge_arrow_pr.py]: It seems that `, ''` is needless.

**Review by kou — APPROVED** [review]
+1

File: dev/merge_arrow_pr.py (hunk 1 above)
Comment (pitrou) [reply dev/merge_arrow_pr.py]: I'm thinking about the case where the environment variable is set but empty. Perhaps I'm just caring too much :-)

Comment (kou) [reply dev/merge_arrow_pr.py]: Yes. I understood what you want.
It works for empty value without `, ''`:

```console
% python3 -c 'import os; print(os.environ.get("ARROW_PROJECT_NAME") or "arrow")'
arrow
% ARROW_PROJECT_NAME= python3 -c 'import os; print(os.environ.get("ARROW_PROJECT_NAME") or "arrow")'
arrow
% ARROW_PROJECT_NAME=my-arrow python3 -c 'import os; print(os.environ.get("ARROW_PROJECT_NAME") or "arrow")'
my-arrow
```

**Review by kszucs — APPROVED** [review]

File: dev/merge_arrow_pr.py (hunk 1 above)
Comment (kszucs) [reply dev/merge_arrow_pr.py]: Updated.

**Pull request closed without merging**